benchmark/array/array_benchmark.cpp

```diff
@@ -0,0 +1,107 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// Copyright (c) 2021 Team Dissolve and contributors
+#include "benchmark/benchmark.h"
+#include "templates/array.h"
+#include "templates/array2d.h"
+#include "templates/array3d.h"
+
+#include <numeric>
+#include <random>
+
+// Benchmark array
+
+// Benchmarks random access into the array structure
+
+static void BM_Array_1d(benchmark::State &state)
+{
+    int bytes = state.range(0);
+    int numVals = bytes / sizeof(int) / 2;
+    std::default_random_engine generator;
+    std::uniform_int_distribution<int> distribution(std::numeric_limits<int>::min(), std::numeric_limits<int>::max());
+    std::vector<int> indicies(numVals);
+    std::iota(indicies.begin(), indicies.end(), 0);
+    std::shuffle(indicies.begin(), indicies.end(), generator);
+    Array<int> array(numVals);
+    for (int i = 0; i < array.size(); ++i)
+    {
+        array[i] = distribution(generator);
+    }
```

Comment:
```suggestion
    std::fill(array.begin(), array.end(), distribution(generator));
```

Okay, switching this over to an algorithm isn't really that important.  However, I was curious about whether the above code was actually correct or if this needs `std::generate`.  Basically, is each element supposed to have an identical distribution or are they supposed to be unique?

Reply:
If the intent is to fill the array with a distribution of random `int`s, then `std::fill` is going to give an array full of the same `int`...

Reply:
Yeah the intention is that all elements of the array are random integers.